src/services/telemetryService.ts

```diff
@@ -3,15 +3,38 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { AppenderData, BaseTelemetryAppender, BaseTelemetryClient, BaseTelemetryReporter, RawTelemetryEventProperties, TelemetryEventProperties } from '../common/telemetry';
-import { Analytics } from '@segment/analytics-node';
+import { AppenderData, BaseTelemetryClient, BaseTelemetryReporter, RawTelemetryEventProperties, TelemetryEventProperties } from '../common/telemetry';
+import { Analytics, AnalyticsSettings } from '@segment/analytics-node';
 import * as os from 'os';
 import * as vscode from 'vscode';
-import { Configuration } from '../configuration';
 
-const analyticsClientFactory = async (key: string, logger: vscode.LogOutputChannel, isProduction: boolean): Promise<BaseTelemetryClient> => {
-	let segmentAnalyticsClient = new Analytics({ writeKey: key });
+const analyticsClientFactory = async (gitpodHost: string, segmentKey: string, logger: vscode.LogOutputChannel): Promise<BaseTelemetryClient> => {
+	const serviceUrl = new URL(gitpodHost);
 
+	const settings: AnalyticsSettings = {
+		writeKey: segmentKey,
+		// in dev mode we report directly to IDE playground source
+		host: "https://api.segment.io",
+		path: "/v1/batch"
+	}
+	if (segmentKey === "untrusted-dummy-key") {
+		settings.host = gitpodHost;
+		settings.path = '/analytics' + settings.path;
+	} else {
+		if (serviceUrl.host !== "gitpod.io" && !serviceUrl.host.endsWith(".gitpod-dev.com")) {
+			logger.warn(`No telemetry: dedicated installations should send data always to own endpoints, host: ${serviceUrl.host}`);
+			return {
+				logEvent: () => { },
+				logException: () => { },
+				flush: () => { },
+			}
+		}
```

Comment:
So if we are running this extension from sources and connect to dedicated, we cannot send telemetry to ide playground for testing and need to switch to gitpod.io :thinking: why is that? I think it should be fine to do so

Reply:
@akosyakov  Yep, `else` should never happen.

I'm not sure about the target:

🅰️ 
1. Source -> Local trace
2. SaaS -> SaaS
3. dev -> dev
4. Others -> their gitpodHost

or 🅱️ 
1. Source -> Local trace
2. SaaS -> SaaS
3. dev -> dev
4. Others -> omit

==============
Right now, we want and are going with 🅰️ . So we can remove `else`?

Reply:
Telemetry from dedicated should always be reported to a corresponding dedicated cell. We are legally not allowed to distribute it anywhere else.

Reply:
So else is working for the case we build from source and use dedicated? ok then